File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
#include <vector>
#include <unordered_map>
#include <string>
#include <iostream>
#include <algorithm>

using namespace std;
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        
        unordered_map<int, int> grps;

        for (auto i: nums) {
            grps[i]++;
        }

        vector<vector<int>> buckets(nums.size() + 1);

        for(const auto&pair: grps) {
            int number = pair.first;
            int frequency = pair.second;

            buckets[frequency].push_back(number);
        }

        vector<int> result;

        for(int i = buckets.size() - 1; i >=0 & result.size() < k; --i) {

            if (!buckets[i].empty()) {
                for (auto ele: buckets[i]) { result.push_back(ele); }
            }

        }

        return result;
    }
};

File: Data Structures & Algorithms/reverse-integer/submission-0.py
class Solution:
    def reverse(self, x: int) -> int:
        
        result = 0
        nums = abs(x)

        while nums > 0:
            result = (result * 10) + (nums % 10)
            nums = nums // 10

        if not (-2**31 <= result <= 2**31-1):
            return 0

        return result if x > 0 else -result